packages/block-editor/src/components/block-toolbar/utils.js

```diff
@@ -38,10 +50,10 @@ export function useDebouncedShowMovers( {
 				clearTimeout( timeout );
 			}
 			if ( ! showMovers ) {
-				setShowMovers( true );
+				handleOnChange( true );
 			}
 		},
-		[ showMovers ]
+		[ handleOnChange, showMovers ]
```

Comment:
Thanks for your feedback!

In this case, the state of `showMovers` works as a guard for both `setShowMovers` and `onChange`. We only want to make the state change if it's different. 

Re: `handleOnChange`. Are you suggesting something like this?

```jsx
const handleOnChange = useCallback(( nextIsFocused ) => {
    setShowMovers( nextIsFocused );
    onChange( nextIsFocused );
}, [ onChange ])
```

And with this, I think we can remove it from being a dependency from the debounced Show/Hide functions

🤔

Reply:
Probably more the opposite, to instead remove `useCallback` from `debouncedShowMovers`. Over time, I've come to see `useCallback` as something of a code smell.

i.e.

```
function debouncedShowMovers( event ) {
	if ( event ) {
		event.stopPropagation();
	}

	const timeout = timeoutRef.current;

	if ( timeout && clearTimeout ) {
		clearTimeout( timeout );
	}
	if ( ! showMovers ) {
		handleOnChange( true );
	}
}
```

Would you see a problem with this alternative?

Reply:
> In this case, the state of `showMovers` works as a guard for both `setShowMovers` and `onChange`. We only want to make the state change if it's different.

But it's only incidentally accounting for changes in `onChange` on account for the fact that `handleOnChange` is a new function each render, which counteracts any benefits we might otherwise get from using `useCallback`. In other words, `debouncedShowMovers` will also be a new reference on every render.

Reply:
> Would you see a problem with this alternative? For what reason do we want useCallback ?

No problems 😊 . I was using `useCallback` as that was something I've seen in many places in our codebase, and it felt like a good idea to continue. But like you've suggested, it may be pre-optimizing, and perhaps causing more trouble and it's worth.

Reply:
The way I've understood it is that `useCallback` is useful if and only if there's an important reason to avoid changing references, and that in all other cases it's not only premature, but often yields _worse_ performance.

Related: https://kentcdodds.com/blog/usememo-and-usecallback

Reply:
KCD saves the day!